packages/flutter/test/material/popup_menu_theme_test.dart

```diff
@@ -0,0 +1,255 @@
+              itemBuilder: (BuildContext context) {
+                final List<PopupMenuEntry<Object>> list = <PopupMenuEntry<Object>>[
+                  PopupMenuItem<void>(
+                    key: popupItemKey,
+                    child: const Text(''),
+                  ),
+                ];
+                return list;
+              },
+            ),
+          ],
+        ),
+      ),
+    ));
+
+    await tester.tap(find.byKey(popupButtonKey));
+    await tester.pumpAndSettle();
+
+    final Material button = tester.widget<Material>(
+      find
+          .descendant(
+            of: find.byKey(popupButtonApp),
+            matching: find.byType(Material),
+          )
+          .last,
+    );
+    expect(button.color, popupMenuEntryTheme.surfaceContainerColor);
+    expect(button.shape, popupMenuEntryTheme.shape);
+    expect(button.elevation, popupMenuEntryTheme.elevation);
+
+    final AnimatedDefaultTextStyle text =
+        tester.widget<AnimatedDefaultTextStyle>(
+      find
+          .descendant(
+            of: find.byKey(popupItemKey),
+            matching: find.byType(AnimatedDefaultTextStyle),
+          )
+          .last,
+    );
+    expect(text.style, popupMenuEntryTheme.textStyle);
+  });
+
+  testWidgets('PopupMenuEntry widget properties take priority over theme',
+      (WidgetTester tester) async {
+    final PopupMenuEntryThemeData popupMenuEntryTheme = _popupMenuEntryTheme();
+    final Key popupButtonKey = UniqueKey();
+    final Key popupButtonApp = UniqueKey();
+    final Key popupItemKey = UniqueKey();
+
+    const Color surfaceContainerColor = Colors.purple;
+    const ShapeBorder shape = RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(9.0)),
+    );
+    const double elevation = 7.0;
+    const TextStyle textStyle = TextStyle(
+        color: Color(0x00000000), textBaseline: TextBaseline.alphabetic);
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(popupMenuEntryTheme: popupMenuEntryTheme),
+      key: popupButtonApp,
+      home: Material(
+        child: Column(
+          children: <Widget>[
+            PopupMenuButton<void>(
+              key: popupButtonKey,
+              elevation: elevation,
+              color: surfaceContainerColor,
+              shape: shape,
+              itemBuilder: (BuildContext context) {
+                final List<PopupMenuEntry<Object>> list = <PopupMenuEntry<Object>>[
+                  PopupMenuItem<void>(
+                    key: popupItemKey,
+                    textStyle: textStyle,
+                    child: const Text(''),
+                  ),
+                ];
+                return list;
+              },
+            ),
+          ],
+        ),
+      ),
+    ));
+
+    await tester.tap(find.byKey(popupButtonKey));
+    await tester.pumpAndSettle();
+
+    final Material button = tester.widget<Material>(
+      find
+          .descendant(
+            of: find.byKey(popupButtonApp),
```

Comment:
I don't think I quite follow how this gets the actual menu widget, could you explain?

Reply:
The last actual piece of material under the popupButtonApp widget is the menu widget itself. So by finding the last descendent of popupButtonApp that is of type Material, this code gets the actual menu widget.